lib/Core/ExecutorUtil.cpp

```diff
@@ -126,6 +126,30 @@ namespace klee {
     if (numOperands > 1) op2 = evalConstant(ce->getOperand(1), ki);
     if (numOperands > 2) op3 = evalConstant(ce->getOperand(2), ki);
 
+    /* Checking for possible errors during constant folding */
+    switch (ce->getOpcode()) {
+    case Instruction::SDiv:
+    case Instruction::UDiv:
+    case Instruction::SRem:
+    case Instruction::URem:
+      if (op2->getLimitedValue() == 0) {
+	std::string msg("Division/modulo by zero during constant folding at location ");
+	llvm::raw_string_ostream os(msg);
+	os << (ki ? ki->printFileLine().c_str() : "[unknown]");
+	klee_error("%s", os.str().c_str());
```

Comment:
There are tradeoffs regarding early and lazy evaluation here, and I think it's better to flag such bugs early on.  

Regarding formatting, we've discussed this so many times, but I'll say again we don't have strict formatting standards, other than being consistent with existing code (which I believe is also the golden rule of the LLVM coding style, which you like).  So if a file already uses tabs, we keep using tabs and the same for spaces.  Since this file seems to be using a mixed style, I'll add a separate commit to convert all tabs to spaces.